### PR TITLE
feat: support global search and switch

### LINKED ISSUE #7: [FR] Add global search and switch support
enter for switching to the file, and command + enter for replace.

### PR DESCRIPTION
**Pull request by quanru**
close #7

### CODE REVIEW

**Comment by MahmoudFawzyKhalil** [comment]
Thank you very much for your contribution, quanru. I have been busy and unable to work on the plugin for a while.

Looking at the code I only see two things I think we should change:
**First change**
I think changing the default of replacing from `Enter` to `Meta Enter` will impact people who already use the plugin negatively. They will need to learn the new way of interacting with it. Also, shift enter is not suitable for mobile users which rely on tapping an entry to replace.

So in order not to impact previous users, I think we should make this a setting in the plugin's settings with a default that is `Enter` to replace and `Meta Enter` to open. This can be changed by the user. The setting can be named something like "Switch replace and open keys".

**Second change**
The code that opens a file and highlights the selection should be moved into `file-operator.tsx`.

**Comment by quanru** [comment]
@MahmoudFawzyKhalil Thank you for replying, this is a search plugin that I have been looking for a long time. I just reverse the function of Enter and Meta Enter, so it will not impact people who already use the plugin. I found there is a feature(Replace all) has not been commit, I will add a toggle setting(Reverse hotkey) to reverse the hot key after the setting tab is commit, is it ok?

File: src/obsidian-components/search-and-replace-modal.tsx [hunk 1]
```diff
@@ -40,15 +40,27 @@ export class SearchAndReplaceModal extends Modal {
 			eventBridge.onArrowDown?.(e, ctx);
 		});
 
-		// Replace note at selectedIndex
+		// Open selectedIndex note
```
Comment (MahmoudFawzyKhalil) [inline src/obsidian-components/search-and-replace-modal.tsx]: Please change this back to "Replace note at selectedIndex"

File: src/obsidian-components/search-and-replace-modal.tsx [hunk 2]
```diff
@@ -40,15 +40,27 @@ export class SearchAndReplaceModal extends Modal {
...
 		this.scope.register([], "Enter", (e, ctx) => {
 			e.preventDefault();
 
 			// Prevent press and hold
 			if (e.repeat) return;
 
+			this.close();
```
Comment (MahmoudFawzyKhalil) [inline src/obsidian-components/search-and-replace-modal.tsx]: Please remove this line as we do not want to close the modal if the action is replace

File: src/obsidian-components/search-and-replace-modal.tsx [hunk 3]
```diff
@@ -40,15 +40,27 @@ export class SearchAndReplaceModal extends Modal {
...
 			eventBridge.onEnter?.(e, ctx);
 		});
+
+		// Replace note at selectedIndex
```
Comment (MahmoudFawzyKhalil) [inline src/obsidian-components/search-and-replace-modal.tsx]: Please change this to Open note at selectedIndex

File: src/obsidian-components/search-and-replace-modal.tsx [hunk 4]
```diff
@@ -40,15 +40,27 @@ export class SearchAndReplaceModal extends Modal {
...
+		this.scope.register(["Meta"], "Enter", (e, ctx) => {
+			e.preventDefault();
+
+			// Prevent press and hold
+			if (e.repeat) return;
+
+			this.close();
```
Comment (MahmoudFawzyKhalil) [inline src/obsidian-components/search-and-replace-modal.tsx]: Please make `this.close()` the last line in the method

**Comment by MahmoudFawzyKhalil** [comment]
Thank you for resolving the comments. I will go ahead and merge and create a new release (0.2.0) now.

**Pull request merged**